Docs: overview of ingest storage architecture (pt. 2)

dimitarvdimitrov opened the pull request:
This adds some missing content to https://github.com/grafana/mimir/pull/12453

i've left a few TODOs and comments around places of the content where i'm not sure about the best way to present some info.

dimitarvdimitrov commented:
I think I agree with you. These details can be a bit overwhelming if someone wants to just get to know Mimir. How should we call the new page - "Ingest storage architecture?" My problem with that is that the lifecycle of a query is broadly the same in classic and ingest architecture. So we either need two topics which would include "the lifecycle of a query" or pull out only the ingest-storage-related bits to a separate topic, which would make the content live in a lot of places and harder to follow.

I'm very likely thinking about this in the wrong way, so I think if you've got an idea, it might be best if you make the change on your branch. I've incorporated all of the wording/style changes, dropped the block builder mention and the config flag mention. We can maybe merge this PR (it's targeting your branch, not main) and continue the work there?

tacole02 commented:
Hello! Actually, I'm just looking at https://github.com/grafana/mimir/pull/12480, and I feel like my thoughts there apply to the architectural overview too. If we say that we support classic architecture in Mimir 3.0, then we also need to document how it works. Perhaps then we don't want to do a straight-up replacement of all of info in the read/write paths. I can try setting this up on my branch. Maybe hold off on merging this until we get the structure figured out?

tacole02 commented:
Hey @dimitarvdimitrov I moved some things around on my branch, so you might want to pull them. This content now belongs at [https://deploy-preview-mimir-12453-zb444pucvq-vp.a.run.app/docs/mimir/v2.17.x/get-started/about-grafan[…]itecture/about-ingest-storage-architecture/](https://deploy-preview-mimir-12453-zb444pucvq-vp.a.run.app/docs/mimir/v2.17.x/get-started/about-grafana-mimir-architecture/about-ingest-storage-architecture/#grafana-mimir-components-in-ingest-storage-architecture) when it's ready.

dimitarvdimitrov commented on docs/sources/mimir/get-started/about-grafana-mimir-architecture/index.md:
```diff
@@ -134,3 +174,28 @@ Grafana Mimir requires any of the following object stores for the block files:
+
+Ingesters do not need Kafka to be available to serve queries. During a Kafka outage, ingesters can continue serving queries with the data they have already consumed from Kafka and stored in memory or on disk.
+
+Kafka retention should be configured based on how long of an ingester outage the operator wants to be able to withstand without data loss. If ingesters are unavailable for longer than the Kafka retention period, they will not be able to resume from the offset in their consumer group, leading to a gap in ingested samples.
```
```suggestion
```

dimitarvdimitrov commented:
@tacole02 sorry for not merging this PR since earlier last week. I was on PTO on Friday and this week has been only on call so far. Feel free to merge this if it's blocking further work

I moved the content I added in this PR to the ingest storage architecture topic.

@narqo thanks for the review, I incorporated your feedback in this PR

narqo reviewed (approved):
🔥
